src/sonarqube.ts

```diff
@@ -1,14 +1,148 @@
+                            throw new Error(
+                                `Could not find Sonarqube component for this file in Sonarqube project "${project}"`
+                            )
+                        }
+                        const branches = await listBranches({ project: component.project, sonarqubeApiUrl })
+                        const branch = branches.find(branch => branch.commit.sha === commitID)
+                        if (!branch) {
+                            console.warn(
+                                `No Sonarqube branch found for commit ID ${commitID}, falling back to default branch`
+                            )
+                        }
+                        const issues = await searchIssues({
+                            sonarqubeApiUrl,
+                            componentKeys: [component.key],
+                            branch: branch?.name,
+                        })
+                        return { editor, issues }
+                    } catch (error) {
+                        console.error(error)
+                        sourcegraph.internal.updateContext({ 'sonarqube.errorMessage': error?.message })
+                        return { editor, issues: [] }
+                    }
+                })
+            )
+            .subscribe(({ editor, issues }) => {
+                editor.setDecorations(
+                    decorationKey,
+                    issues.map(issue => {
+                        const { effort, type, severity, message, key, project } = issue
+                        const { startLine, endLine, startOffset, endOffset } = issue.textRange
+                        const dateString = new Date(issue.creationDate).toLocaleDateString()
+                        const tagsString = issue.tags.map(tag => `🏷 ${tag}`).join(' ')
+                        const typeString = type.replace(/_/g, ' ')
+                        const severityIcon = severityIcons[severity]
+                        const typeIcon = typeIcons[type]
+                        return {
+                            range: new sourcegraph.Range(startLine - 1, startOffset, endLine - 1, endOffset),
+                            backgroundColor: 'rgba(var(--oc-red-7-rgb), 0.1)',
+                            border: '1px solid rgba(var(--oc-red-7-rgb), 0.9)',
+                            after: {
+                                color: 'var(--danger)',
+                                contentText: ` ${typeIcon} ${typeString}: ${message} ${severityIcon} ${severity} ⏱ ${effort} effort 🗓 created ${dateString} ${tagsString} `,
+                                linkURL: new URL(`/project/issues?id=${project}&open=${key}`, sonarqubeUrl).href,
+                            },
+                        }
+                    })
+                )
+            })
+    )
+
+    context.subscriptions.add(
+        sourcegraph.commands.registerCommand('sonarqube.toggleIssuesOnCodeViews', async () => {
```

Comment:
Is this dead code? I only see a built-in command in the manifest

Reply:
Yep, thanks!